src/main/java/com/scylladb/cdc/debezium/connector/ScyllaExtractSimpleNewRecordState.java

```diff
@@ -0,0 +1,83 @@
+package com.scylladb.cdc.debezium.connector;
+
+import io.debezium.transforms.ExtractNewRecordState;
+
+import org.apache.kafka.common.cache.Cache;
+import org.apache.kafka.common.cache.LRUCache;
+import org.apache.kafka.common.cache.SynchronizedCache;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.transforms.util.SchemaUtil;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.data.Schema.Type;
+
+import static org.apache.kafka.connect.transforms.util.Requirements.requireStruct;
+
+public class ScyllaExtractSimpleNewRecordState<R extends ConnectRecord<R>> extends ExtractNewRecordState<R> {
```

Comment:
Maybe such a name would be better and simpler?

```suggestion
public class ScyllaExtractNewRecordState<R extends ConnectRecord<R>> extends ExtractNewRecordState<R> {
```